pkg/controllers/routing/network_routes_controller.go

```diff
@@ -1084,106 +1088,22 @@ func (nrc *NetworkRoutingController) startBgpServer(grpcServer bool) error {
+		if peerCfgs == nil {
 			return nil
 		}
-		ipStrings := stringToSlice(nodeBgpPeersAnnotation, ",")
-		peerIPs, err := stringSliceToIPs(ipStrings)
-		if err != nil {
-			err2 := nrc.bgpServer.StopBgp(context.Background(), &gobgpapi.StopBgpRequest{})
-			if err2 != nil {
-				klog.Errorf("Failed to stop bgpServer: %s", err2)
-			}
-
-			return fmt.Errorf("failed to parse node's Peer Addresses Annotation: %s", err)
-		}
-
-		// Get Global Peer Router ASN configs
-		nodeBgpPeerPortsAnnotation, ok := node.Annotations[peerPortAnnotation]
-		// Default to default BGP port if port annotation is not found
-		var peerPorts = make([]uint32, 0)
-		if ok {
-			portStrings := stringToSlice(nodeBgpPeerPortsAnnotation, ",")
-			peerPorts, err = stringSliceToUInt32(portStrings)
-			if err != nil {
-				err2 := nrc.bgpServer.StopBgp(context.Background(), &gobgpapi.StopBgpRequest{})
-				if err2 != nil {
-					klog.Errorf("Failed to stop bgpServer: %s", err2)
-				}
-				return fmt.Errorf("failed to parse node's Peer Port Numbers Annotation: %s", err)
-			}
-		}
-
-		// Get Global Peer Router Password configs
-		var peerPasswords []string
-		nodeBGPPasswordsAnnotation, ok := node.Annotations[peerPasswordAnnotation]
-		if !ok {
-			klog.Infof("Could not find BGP peer password info in the node's annotations. Assuming no passwords.")
-		} else {
-			passStrings := stringToSlice(nodeBGPPasswordsAnnotation, ",")
-			peerPasswords, err = stringSliceB64Decode(passStrings)
-			if err != nil {
-				err2 := nrc.bgpServer.StopBgp(context.Background(), &gobgpapi.StopBgpRequest{})
-				if err2 != nil {
-					klog.Errorf("Failed to stop bgpServer: %s", err2)
-				}
-				return fmt.Errorf("failed to parse node's Peer Passwords Annotation")
-			}
-		}
-
-		// Get Global Peer Router LocalIP configs
-		var peerLocalIPs []string
-		nodeBGPPeerLocalIPs, ok := node.Annotations[peerLocalIPAnnotation]
-		if !ok {
-			klog.Infof("Could not find BGP peer local ip info in the node's annotations. Assuming node IP.")
-		} else {
-			peerLocalIPs = stringToSlice(nodeBGPPeerLocalIPs, ",")
-			err = func() error {
-				for _, s := range peerLocalIPs {
-					if s != "" {
-						ip := net.ParseIP(s)
-						if ip == nil {
-							return fmt.Errorf("could not parse \"%s\" as an IP", s)
-						}
-					}
-				}
-
-				return nil
-			}()
-			if err != nil {
-				err2 := nrc.bgpServer.StopBgp(context.Background(), &gobgpapi.StopBgpRequest{})
-				if err2 != nil {
-					klog.Errorf("Failed to stop bgpServer: %s", err2)
-				}
-
-				return fmt.Errorf("failed to parse node's Peer Local Addresses Annotation: %s", err)
-			}
-		}
 
 		// Create and set Global Peer Router complete configs
-		nrc.globalPeerRouters, err = newGlobalPeers(peerIPs, peerPorts, peerASNs, peerPasswords, peerLocalIPs,
+		nrc.globalPeerRouters, err = newGlobalPeers(peerCfgs.RemoteIPs(), peerCfgs.Ports(), peerCfgs.RemoteASNs(), peerCfgs.Passwords(), peerCfgs.LocalIPs(),
```

Comment:
We could probably consolidate this to make it just accept the peerCfg instead of piece-mealing it out like this.

Reply:
Hows this?

Reply:
Much better!